src/serialization/interval_log/mod.rs

```diff
@@ -170,69 +169,138 @@ use nom::{double, line_ending, not_line_ending, IResult};
 use super::super::{Counter, Histogram};
 use super::Serializer;
 
-/// Write headers for an interval log.
+/// Prepare an `IntervalLogWriter`.
 ///
 /// This type only allows writing comments and headers. Once you're done writing those things, use
 /// `into_log_writer()` to convert this into an `IntervalLogWriter`.
-pub struct IntervalLogHeaderWriter<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> {
-    internal_writer: InternalLogWriter<'a, 'b, W, S>,
+pub struct IntervalLogWriterBuilder {
+    comments: Vec<String>,
+    start_time: Option<f64>,
+    base_time: Option<f64>,
+    max_value_divisor: f64,
 }
 
-impl<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> IntervalLogHeaderWriter<'a, 'b, W, S> {
+impl IntervalLogWriterBuilder {
     /// Create a new log writer that writes to `writer` and serializes histograms with `serializer`.
-    pub fn new(writer: &'a mut W, serializer: &'b mut S) -> IntervalLogHeaderWriter<'a, 'b, W, S> {
-        IntervalLogHeaderWriter {
-            internal_writer: InternalLogWriter {
-                writer,
-                serializer,
-                text_buf: String::new(),
-                serialize_buf: Vec::new(),
-            },
+    pub fn new() -> IntervalLogWriterBuilder {
+        IntervalLogWriterBuilder {
+            comments: Vec::new(),
+            start_time: None,
+            base_time: None,
+            max_value_divisor: 1.0,
         }
     }
 
-    /// Add a comment line.
+    /// Add a comment line to be written when the writer is built.
     ///
     /// If you do silly things like write a comment with a newline character, you'll end up with
     /// an un-parseable log file. Don't do that.
-    pub fn write_comment(&mut self, s: &str) -> io::Result<()> {
-        self.internal_writer.write_comment(s)
+    pub fn add_comment(&mut self, s: &str) -> &mut Self {
+        self.comments.push(s.to_owned());
+        self
     }
 
-    /// Write a StartTime log line. See the module-level documentation for more info.
+    /// Set a StartTime. See the module-level documentation for more info.
     ///
-    /// This should only be called once to avoid creating a confusing interval log.
-    pub fn write_start_time(&mut self, seconds_since_epoch: f64) -> io::Result<()> {
-        self.internal_writer.write_fmt(format_args!(
-            "#[StartTime: {:.3} (seconds since epoch)]\n",
-            seconds_since_epoch
-        ))
+    /// This can be called multiple times, but only the value for the most recent invocation will
+    /// be written.
+    pub fn with_start_time(&mut self, seconds_since_epoch: f64) -> &mut Self {
```

Comment:
Hmm... perhaps. That would mean we'd have to do something about handling the error case for `duration_since()`, though. We could just take the `Duration` instead of the `SystemTime` to push that decision off to the caller, but still have something more "time-flavored" than a lowly `f64`.

Reply:
```rust
let since_epoch = match time.duration_since(UNIX_EPOCH) {
    Ok(d) => dur_to_f64(d),
    Err(ste) => -1 * dur_to_f64(ste.duration())
};
```
?

Reply:
Yeah, that should work, and we're very unlikely to hit the Err case, though I wonder if maybe we should panic because your timekeeping is clearly hosed if your system thinks it's before 1970?

Reply:
Doh -- if we accept `SystemTime`, we can't have repeatable tests because you can't construct an arbitrary `SystemTime`.

Reply:
Actually, you can, by adding to `UNIX_EPOCH` ;)
As for panicking, I'd rather avoid it given that we don't have to.

Reply:
Ah, good point. In that case I'm 👍 because I can get my testing done *and* we can always provide another way to set the times if `SystemTime` turns out to be awkward for users.